src/TraitsV2/TaClassCompositionElement.class.st

```diff
@@ -22,9 +22,9 @@ TaClassCompositionElement >> selectors [
 
 { #category : #accessing }
 TaClassCompositionElement >> slots [
-	^ (innerClass slots
-		reject: [ :e | TraitedClass slots anySatisfy: [ :x | x name = e name ] ])
-		collect: [ :e | self copySlot: e ]
+	^ innerClass slots
+		reject: [ :e | TraitedClass slots anySatisfy: [ :x | x name = e name ] ]
```

Comment:
Nowadays we don't have instance variables in the TraitedClass, but removing it will limit the future ability. Maybe we should rewrite the method to not do the reject if the TraitedClass slots is empty.

Reply:
But what about my scenario in case when TraitedClass has var:
>>And what will happens if I will have class with var names equal to one from TraitedClass? And then I will add trait to it.

Is it normal conflict as other trait conflicts? Then resolving it with alias will break this rejection logic. Does it?
Should not it be like:
```Smalltalk
innerClass slots reject: [:e | TraitedClass slots includes: e originSlot] "like originMethod"
```
But I don't know what is innerClass here. So maybe it is all wrong

Reply:
Could you explain why TraitedClass slots should be rejected?

Reply:
Because they are handled in different way. 
Basically, including them depends if they are in the superclass or not. As this trait is included in all the traited class, if I inherit from a traited class (the slots and methods) are already in the superclass and should not be duplicated. However, if the new class inherits from a normal class the methods and slots should be added.

Reply:
Ok. I see complexity.
But do you agree that checking slots by name can lead to problems?